public/assets/cpu.js

```diff
@@ -82,14 +82,24 @@ function CPU(memamt) {
     this.keyLoad = 'load';
     this.keyMem = 'memoryOrder';
 
-    this.instruction_amounts = {
+    // stores the current num of instructions in one clock cycle
+    this.curr_instructions = {
         'arithmetic' : 0,
         'controlTransfer' : 0,
         'store' : 0,
         'load' : 0,
         'memoryOrder' : 0
     }
 
+    // stores the exponential rolling average of each type of instr
+    this.instruction_amounts = {
+        'arithmetic' : [0, 0],
+        'controlTransfer' : [0, 0],
+        'store' : [0, 0],
+        'load' : [0, 0],
```

Comment:
what if instead of having this 2 element array, you have a nested object?

i see down below how you are using it by indexing 0 or 1.

what if you did a model like

```
const myObj = {
  load: {
    myKey: 'myValue'
  },
  ...
}

// then you access it like this
myObj.load.myKey++; // for example
```

Reply:
I changed it to this, however, when i iterate over the object, I have to using indexing by key to actually get the value. But this reduces the number of (accesses by index) by one for each key at least